### PR TITLE
8346151: Add transformer error logging to VerifyLocalVariableTableOnRetransformTest

### PR DESCRIPTION
**Pull request by alexmenkov**
In some circumstances ClassFileTransformer.transform can get ClassCircularityError or LinkageError concatenating strings (see JDK-8264667, JDK-8262002).
VerifyLocalVariableTableOnRetransformTest fails sometimes on Oracle CI.
The fix adds handling of the errors to get information for analysis.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8346151](https://bugs.openjdk.org/browse/JDK-8346151): Add transformer error logging to VerifyLocalVariableTableOnRetransformTest (**Sub-task** - P4)

### Reviewers
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**)
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22727/head:pull/22727` \
`$ git checkout pull/22727`

Update a local copy of the PR: \
`$ git checkout pull/22727` \
`$ git pull https://git.openjdk.org/jdk.git pull/22727/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22727`

View PR using the GUI difftool: \
`$ git pr show -t 22727`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22727.diff">https://git.openjdk.org/jdk/pull/22727.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22727#issuecomment-2540397933)
</details>

### CODE REVIEW

File: test/jdk/java/lang/instrument/VerifyLocalVariableTableOnRetransformTest.java [hunk 1]
```diff
@@ -247,8 +250,20 @@ private void saveMismatchedBytes(byte[] classfileBuffer) {
...
-                + "' of " + classfileBuffer.length + " bytes.");
+            // String concatenation can cause LinkageError or ClassCircularityError (see JDK-8311534).
+            // Need to log it for analysis.
+            try {
+                System.out.println(this + ".transform() sees '" + className
+                    + "' of " + classfileBuffer.length + " bytes.");
+            } catch (Throwable t) {
+                // try to log, save the error for handling by main thread if the logging fails
+                try {
+                    transformerException.printStackTrace();
```
Comment (plummercj) [inline test/jdk/java/lang/instrument/VerifyLocalVariableTableOnRetransformTest.java:261]: Isn't this suppose to be `t.printStackTrace()`?

File: test/jdk/java/lang/instrument/VerifyLocalVariableTableOnRetransformTest.java [hunk 2]
```diff
@@ -247,8 +250,20 @@ private void saveMismatchedBytes(byte[] classfileBuffer) {
...
 
-            System.out.println(this + ".transform() sees '" + className
-                + "' of " + classfileBuffer.length + " bytes.");
+            // String concatenation can cause LinkageError or ClassCircularityError (see JDK-8311534).
+            // Need to log it for analysis.
+            try {
+                System.out.println(this + ".transform() sees '" + className
+                    + "' of " + classfileBuffer.length + " bytes.");
+            } catch (Throwable t) {
+                // try to log, save the error for handling by main thread if the logging fails
```
Comment (plummercj) [inline test/jdk/java/lang/instrument/VerifyLocalVariableTableOnRetransformTest.java:259]:
```suggestion
                // Try to log. Save the error for handling by main thread if the logging fails.
```

**Review by plummercj — CHANGES_REQUESTED** [review]

File: test/jdk/java/lang/instrument/VerifyLocalVariableTableOnRetransformTest.java (hunk 1 above)
Comment (alexmenkov) [reply test/jdk/java/lang/instrument/VerifyLocalVariableTableOnRetransformTest.java]: Good catch!
Fixed.

File: test/jdk/java/lang/instrument/VerifyLocalVariableTableOnRetransformTest.java (hunk 2 above)
Comment (alexmenkov) [reply test/jdk/java/lang/instrument/VerifyLocalVariableTableOnRetransformTest.java]: Done

**Review by plummercj — APPROVED** [review]
Copyright needs updating. Otherwise looks good.

**Review by plummercj — APPROVED** [review]

**Review by sspitsyn — APPROVED** [review]
This is good.

**Pull request closed without merging**